File: backend/src/main/java/br/com/adoteme/model/entity/Cliente.java
package br.com.adoteme.model.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Table(name = "Cliente")
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
public class Cliente {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "Codigo")
    private Integer codigo;

    @Column(name = "Nome")
    private String nome;

    @Column(name = "Email")
    private String email;

    @Column(name = "Senha")
    private String senha;

    @Column(name = "Documento")
    private String documento;

    @Column(name = "Telefone")
    private String telefone;

    @Column(name = "Logradouro")
    private String logradouro;

    @Column(name = "Complemento")
    private String complemento;

    @Column(name = "Cidade")
    private String cidade;

    @Column(name = "Bairro")
    private String bairro;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "IdEstado")
    private Estado estado;

    @Column(name = "Cep")
    private String cep;

    @Column(name = "Status")
    private Boolean status;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "IdRole")
    private Role role;

}

File: frontend/src/app/shared/core/stores/cliente-store/cliente.ts
export default class Cliente {
  codigo: number;
  nome: string;
  email: string;
  senha: string;
  documento: string;
  telefone: string;
  logradouro: string;
  complemento: string;
  cep: string;
  cidade: string;
  estado: number;
  bairro: string;

  constructor(cliente?: any) {
    this.codigo = cliente?.codigo ?? 0;
    this.nome = cliente?.nome ?? '';
    this.email = cliente?.email ?? '';
    this.senha = cliente?.senha ?? '';
    this.documento = cliente?.documento ?? '';
    this.telefone = cliente?.telefone ?? '';
    this.logradouro = cliente?.logradouro ?? '';
    this.complemento = cliente?.complemento ?? '';
    this.cep = cliente?.cep ?? '';
    this.cidade = cliente?.cidade ?? '';
    this.estado = cliente?.estado ?? 0;
    this.bairro = cliente?.bairro ?? '';
  }
}
